plugins/nanospell/plugin.js

```diff
@@ -664,19 +664,17 @@
 				var blockList = event.data.blockList;
 				var url = resolveAjaxHandler();
 				var callback = function (data) {
-					var bookmarks;
+					var selectionStart = editor.getSelection().getStartElement();
+					var rootElement;
 					parseRpc(data, words);
-					editor.lockSelection();
-					bookmarks = editor.getSelection().createBookmarks(true);
+
 					for (var i = 0; i < blockList.length; i++) {
-						var rootElement = blockList[i];
+						rootElement = blockList[i];
 						editor.fire(EVENT_NAMES.START_RENDER, {
 							root: rootElement,
-							needsBookmarkCreated: false,
+							needsBookmarkCreated: selectionStart ? rootElement.contains(selectionStart) || rootElement.equals(selectionStart) : null,
```

Comment:
My guess is that `rootElement.contains` is a (relatively) slow call because it needs to walk the DOM. Since we'll be doing this once per block, that's something to avoid if we can (without incurring a big complexity cost).

I think we'd be better off by walking up the tree from `selectionStart` until we hit a node that's the root of a block. We can do that one time, outside the loop, and then the logic within the loop becomes much faster (simple equality check).